### PR TITLE
Expression is not will be quoted

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️/❌
| Tests pass?   | ✔️
| Fixed issues  | #18134

### CODE REVIEW

**Comment by darkdef** [comment]
@rob006 @samdark 
Please review my commit (it's fix instead of #18142)

File: tests/framework/db/QueryBuilderTest.php [hunk 1]
```diff
@@ -2523,6 +2523,17 @@ public function testBuildLikeCondition($condition, $expected, $expectedParams)
         $this->assertEquals($expectedParams, $params);
     }
 
+    /**
+     * @see https://github.com/yiisoft/yii2/issues/18134
+     */
+    public function testExpressionIsNotQuotedInColumnName()
```
Comment (rob006) [inline tests/framework/db/QueryBuilderTest.php:2529]: I would move this test to `likeConditionProvider()`.

**Review by rob006 — APPROVED** [review]

File: tests/framework/db/QueryBuilderTest.php (hunk 1 above)
Comment (darkdef) [reply tests/framework/db/QueryBuilderTest.php]: moved

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

**Pull request merged**